### ISSUE TITLE
ExecWatch doesn't support execute a command and collect its std out/err

### ISSUE THREAD

**Issue by quickwind** [fabric8io/kubernetes-client#728]
Our use case is to copy out file(s) from pod container using tar, which is:
- execute "tar cf - -C / dirToCopy" via ExecWatch
- read output and dump to a tar file
- if failed, collect errors from std error

Following is the example code to show the problem:
```
...
public class ExecPipesExample2 {
    private static final Logger LOGGER = LoggerFactory
            .getLogger(ExecPipesExample2.class);

    public static void main(String[] args)
            throws InterruptedException, IOException {
        String master = "https://yourserver/";

        Config config = new ConfigBuilder().withMasterUrl(master).build();
        KubernetesClient client = new DefaultKubernetesClient(config);

        final PipedInputStream stdoutInput = new PipedInputStream();
        final PipedOutputStream stdoutOutput = new PipedOutputStream();
        final PipedInputStream stderrInput = new PipedInputStream();
        final PipedOutputStream stderrOutput = new PipedOutputStream();
        final AtomicBoolean failed = new AtomicBoolean(false);

        stdoutInput.connect(stdoutOutput);
        stderrInput.connect(stderrOutput);

        try (final ExecWatch watch = client.pods().inNamespace("test")
                .withName("test").inContainer("c1")
                .writingOutput(stdoutOutput).writingError(stderrOutput)
                .usingListener(new ExecListener() {

                    @Override
                    public void onOpen(Response response) {
                        LOGGER.info("onOpen " + response);
                    }

                    @Override
                    public void onFailure(Throwable t, Response response) {
                        try {
                            failed.set(true);
                            stdoutOutput.close();
                            stderrOutput.close();
                        } catch (IOException e) {
                            LOGGER.error("Failed to close stdout and stderr pipes.", e);
                        }
                        LOGGER.error(
                                "onFailure " + t.getMessage() + " " + response);
                    }

                    @Override
                    public void onClose(int code, String reason) {
                        try {
                            stdoutOutput.close();
                            stderrOutput.close();
                        } catch (IOException e) {
                            LOGGER.error("Failed to close stdout and stderr pipes.", e);
                        }
                        LOGGER.info("onClose " + code + " " + reason);
                    }
                }).exec("tar", "cf", "-", "-C", "/", "data");)

        {
            try {
                int length;
                byte[] buffer = new byte[1024];

                while (!Thread.currentThread().isInterrupted()
                        && (length = stdoutInput.read(buffer)) != -1) {
                    
                    byte[] content = new byte[length];
                    System.arraycopy(buffer, 0, content, 0, length);
                    FileUtils.writeByteArrayToFile(new File("response.tar"),
                            content, true);
                }

                while (!Thread.currentThread().isInterrupted()
                        && (length = stderrInput.read(buffer)) != -1) {
                    LOGGER.error(new String(buffer, 0, length));

                }
            } catch (IOException e) {
                if (!Thread.currentThread().isInterrupted()) {
                    LOGGER.error("Error while pumping stream.", e);
                } else {
                    LOGGER.error("Interrupted while pumping stream.", e);
                }
            }
            
        } catch (Exception e) {
            throw KubernetesClientException.launderThrowable(e);
        }

        if (failed.get()) {
            LOGGER.error("Failed");
        } else {
            LOGGER.info("Finished");
        }
    }

}
```

It doesn't work because .usingListener() method will always set withTty to true, and you have no way to set it to false again, why is this? 

Or what should be the way to achive this?

**Comment by quickwind** [comment]
Created PR to change usingListener method to not always set withTTY to true. Please review.

**Comment by rohanKanojia** [comment]
Closed via #729
